src/coreclr/interpreter/compiler.cpp

```diff
@@ -2833,7 +2834,24 @@ void InterpCompiler::EmitCall(CORINFO_RESOLVED_TOKEN* pConstrainedToken, bool re
 
     if (newObjThisArgLocation != INT_MAX)
     {
-        ctorType = GetInterpType(m_compHnd->asCorInfoType(resolvedCallToken.hClass));
+        const char* namespaceName = nullptr;
```

Comment:
I don't see JIT making checks for class names like this. I can see it is checking callInfo.classFlags & CORINFO_FLG_VALUECLASS, I wonder if using such a check would work for us. I may be completely off though.

Reply:
`CORINFO_FLG_VALUECLASS` won't work. It is set for all primitive types and all value types.

@dotnet/jit-contrib How does the JIT figures out the non-standard handling calling convention for RuntimeMethodHandleInternal and similar types? These types are wrapper structs, but they are passed around as their underlying `IntPtr` type. Or does this mismatch happen to not matter the architectures that are currently supported?

Reply:
There is no special handling for these structs in the Jit. All managed calling conventions pass such structs as if they were primitives. Almost all unmanaged ones do as well, except Unix x86 where structs are always _returned_ by-ref (so methods like https://github.com/dotnet/runtime/blob/0ce54a5269102f8e587ec1f6569d7049a5da3af8/src/coreclr/System.Private.CoreLib/src/System/Reflection/MethodBase.CoreCLR.cs#L40 should be broken on Unix x86).

Reply:
I also think it would be good to first try replacing `ctorType == InterpTypeVT` with `ctorType != InterpTypeO`. I believe it should work.